packages/open-truss/src/signals/index.ts

```diff
@@ -21,19 +36,25 @@ export function getSignalsType(
   }
 }
 
+function isSignalLike(obj: unknown): obj is Signal {
+  return typeof obj === 'object' && obj !== null && 'value' in obj
+}
+
 export function SignalType<T>(
   name: string,
-  defaultValue: T,
-  validator?: () => boolean,
-): z.ZodDefault<z.ZodType<Signal<T>>> {
-  validator ||= () => true
+  valueShape: ValueShape<T>,
+): SignalsZodType<T> {
+  const defaultValue = valueShape.parse(undefined)
+  const validator = (val: unknown): boolean => {
+    return isSignalLike(val) && valueShape.parse(val.value) !== undefined
+  }
```

Comment:
Now that we have a zod shape for the signal's value, we can use it to validate that the correct values are passed in 🎉